R/helpers-general.R

```diff
@@ -165,3 +166,124 @@ return_start_end_dates <- function(season) {
     end_date = end_date
   ))
 }
+
+#' Check if a team is valid for afl website
+#'
+#' @param team Team
+#'
+#' @keywords internal
+#' @noRd
+parse_fitzroy_match_id <- function(df, source = "AFL") {
+  source <- check_source(source)
+
+  round_mapping <- round_mapping
+
+  if (source == "AFL") {
+    df <- df %>%
+      dplyr::mutate(utcStartDate = lubridate::as_date(lubridate::ymd_hms(.data$utcStartTime))) %>%
+      dplyr::left_join(round_mapping, dplyr::join_by(x$utcStartDate >= y$start, x$utcStartDate <= y$end)) %>%
+      dplyr::mutate(
+        fitzroy_match_id =
+          glue::glue(
+            "{season_map}_",
+            "{round_map}_",
+            "{parse_team_abbr(home.team.name)}_",
+            "{parse_team_abbr(away.team.name)}"
+          )
+      )
+  }
+  if (source == "footywire") {
+    df <- df %>%
+      dplyr::left_join(round_mapping, dplyr::join_by(x$Date >= y$start, x$Date <= y$end)) %>%
+      dplyr::mutate(
+        fitzroy_match_id =
+          glue::glue(
+            "{season_map}_",
+            "{round_map}_",
+            "{parse_team_abbr(home.team.name)}_",
+            "{parse_team_abbr(away.team.name)}"
+          )
+      )
+  }
+  if (source == "afltables") {
+    df <- df %>%
+      dplyr::left_join(round_mapping, 
+                       dplyr::join_by(x$Date >= y$start, x$Date <= y$end)) %>%
+      dplyr::mutate(
+        fitzroy_match_id =
+          glue::glue(
+            "{season_map}_",
+            "{round_map}_",
+            "{parse_team_abbr(Home.team)}_",
+            "{parse_team_abbr(Away.team)}"
+          )
+      )
+  }
+  if (source == "squiggle") {
+    df <- df %>%
+      dplyr::left_join(round_mapping, dplyr::join_by(x$date >= y$start, x$date <= y$end)) %>%
+      dplyr::mutate(
+        fitzroy_match_id =
+          glue::glue(
+            "{season_map}_",
+            "{round_map}_",
+            "{parse_team_abbr(hteam)}_",
+            "{parse_team_abbr(ateam)}"
+          )
+      )
+  }
+  if (source == "fryzigg") {
+    df <- df %>%
+      dplyr::left_join(round_mapping, dplyr::join_by(x$match_date >= y$start, x$match_date <= y$end)) %>%
+      dplyr::mutate(
+        fitzroy_match_id =
+          glue::glue(
+            "{season_map}_",
+            "{round_map}_",
+            "{parse_team_abbr(match_home_team)}_",
+            "{parse_team_abbr(match_away_team)}"
+          )
+      )
+  }
+
+  df <- df %>%
+    dplyr::select(.data$fitzroy_match_id, dplyr::everything())
+
+  return(df)
+}
+
+#' Internal function to ensure names match between different sources and also name changes.
+#' This gets applied to any web scraper
+#' @param team_name Team name
+#' @export
```

Comment:
Just add the following to the Roxygen to fix the error in the pkgdown github action

```
#' @keywords internal
#' @noRd
```

Can see you've done that in other functions so assume you just missed it, but see other functions in this file if you need examples